subsys/usb/device_next/usbd_ch9.c

```diff
@@ -1132,6 +1132,7 @@ int usbd_handle_ctrl_xfer(struct usbd_context *const uds_ctx,
 		buf, bi->ep, buf->len, bi->setup, bi->data, bi->status);
 
 	if (bi->setup && bi->ep == USB_CONTROL_EP_OUT) {
+		struct udc_data *data = uds_ctx->dev->data;
```

Comment:
The stack must not modify any UDC driver data. I do not know yet why it is required here, need to look at it closer and try out, perhaps next week.